### PR TITLE
Add :steps to releases

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/mix/lib/mix/release.ex [hunk 1]
```diff
@@ -263,6 +270,21 @@ defmodule Mix.Release do
     |> Keyword.put(:kernel, :permanent)
   end
 
+  defp validate_steps!(steps) do
+    if not is_list(steps) or Enum.any?(steps, &(&1 != :assemble and not is_function(&1, 1))) do
+      Mix.raise(
+        "The :steps option must be a list of anonymous functions that receive one argument " <>
```
Comment (fertapric) [inline lib/mix/lib/mix/release.ex]: To remove ambiguity, what do you think about?

```
The :steps option must be a list of:

  * anonymous function that receives one argument
  * the atom :assemble

Got: #{inspect(steps)}
```

File: lib/mix/lib/mix/release.ex [hunk 2]
```diff
@@ -263,6 +270,21 @@ defmodule Mix.Release do
...
+          "or the atom :assemble, got: #{inspect(steps)}"
+      )
+    end
+
+    if :assemble not in steps do
```
Comment (fertapric) [inline lib/mix/lib/mix/release.ex]: Is it ok if the list contains the `:assemble` atom more than once?

File: lib/mix/lib/mix/release.ex (hunk 1 above)
Comment (josevalim) [reply lib/mix/lib/mix/release.ex]: Excellent! :+1:

File: lib/mix/lib/mix/release.ex (hunk 2 above)
Comment (josevalim) [reply lib/mix/lib/mix/release.ex]: Excellent! :+1:

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**